saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/metadata/SAMLExtensionsParser.java

```diff
@@ -28,12 +27,12 @@
  *
  * @author hmlnarik
  */
-public class SAMLExtensionsParser extends AbstractStaxSamlParser<ExtensionsType> {
+public class SAMLExtensionsParser extends AbstractStaxSamlMetadataParser<ExtensionsType> {
```

Comment:
The `SAMLExtensionsParser` should be on par with `protocol.SAMLExtensionsParser`, namely it should `addExtension` instead of `setElement` when processing subelements. This will need work on `org.keycloak.dom.saml.v2.metadata.ExtensionsType` too - it needs to be the same as `org.keycloak.dom.saml.v2.protocol.ExtensionsType`, and its current methods should be `@deprecated` in Javadoc stating they're being obsoleted by `getAny()` / `addExtension()` method. and their implementation changed to:
- `Element element` should be replaced by `List<Object> any`
- getElement should return first element of `any` property or `null` if `any` is empty
- setElement should unconditionally remove all elements from `any` and add a given element as the only one in the list`